File: leetcode-cn/0806.0_Number_of_Lines_To_Write_String.py
'''
执行用时：40 ms, 在所有 Python3 提交中击败了36.59% 的用户
内存消耗：14.8 MB, 在所有 Python3 提交中击败了96.38% 的用户
通过测试用例：27 / 27
'''
class Solution:
    def numberOfLines(self, widths: List[int], s: str) -> List[int]:
        total, last = 1, 0
        for ch in s:
            w = widths[ord(ch) - ord('a')]
            if last + w > 100:
                total += 1
                last = w
            else:
                last += w

        return [total, last]
        

File: java/0806.0_Number_of_Lines_To_Write_String.java
/*
执行用时：0 ms, 在所有 Java 提交中击败了100.00% 的用户
内存消耗：39.5 MB, 在所有 Java 提交中击败了40.89% 的用户
通过测试用例：27 / 27
*/
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        int[] ans = new int[2];
        ans[0] = 1;
        // int[] ans = {1, 0};
        for (char ch: s.toCharArray()) {
            ans[1] += widths[ch - 'a'];
            if (ans[1] > 100) {
                ans[1] = widths[ch - 'a'];
                ans[0]++;
            }
        }
        return ans;
    }
}



/*
执行用时：0 ms, 在所有 Java 提交中击败了100.00% 的用户
内存消耗：39.5 MB, 在所有 Java 提交中击败了39.88% 的用户
通过测试用例：27 / 27
*/
class Solution {
    public int[] numberOfLines(int[] widths, String s) {
        // int[] ans = new int[2];
        // ans[0] = 1;
        int[] ans = {1, 0};
        for (char ch: s.toCharArray()) {
            ans[1] += widths[ch - 'a'];
            if (ans[1] > 100) {
                ans[1] = widths[ch - 'a'];
                ans[0]++;
            }
        }
        return ans;
    }
}
